src/components/icons/event-icon/event-icon.tsx

```diff
@@ -0,0 +1,419 @@
+import { useEffect, useState } from 'react';
+import { thumbnailStyles } from '../../card/styles';
+
+const PILL_PADDING = 13;
+const RIGHT_MARGIN = 16;
+const ICON_SIZE = 180;
+
+const EventIcon = ({ text }: { text?: string }) => {
+    const [textWidth, setTextWidth] = useState(0);
+    const [textEl, setTextEl] = useState<SVGElement | null>();
```

Comment:
yeah I think originally I had done it this way because setting a ref doesn't automatically re-render the component and I needed it to do that, but since now I also have the text width being set as a state variable, I can probably swap this over to a ref.

Reply:
Ah got it thank you, learned something new today!